src/main/java/ArrayStack.java

```diff
@@ -5,5 +5,16 @@
     private int top;
 
 
+    public ArrayStack() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public ArrayStack(int size) {
+        top = 0;
+        // Generic types cannot be instantiated so we cast
+        // This does generate a compile time warning
+        stack = (T[]) new Object[size];
```

Comment:
Just a random run fact, the jdk actually casts the object during the return instead of during creation. [get](https://github.com/openjdk/jdk/blob/bb3d226a4eedb3d3ef73ef91683f45f1db43a74e/src/java.base/share/classes/java/util/ArrayList.java#L409-L412) and [create](https://github.com/openjdk/jdk/blob/bb3d226a4eedb3d3ef73ef91683f45f1db43a74e/src/java.base/share/classes/java/util/ArrayList.java#L156). The net effect is you only have to write the cast in one place, since you can't grow/chrink the array without re-casting it to `T[]` ([grow impl](https://github.com/openjdk/jdk/blob/bb3d226a4eedb3d3ef73ef91683f45f1db43a74e/src/java.base/share/classes/java/util/ArrayList.java#L237-L239)).

Reply:
That is a run fact.